src/utils.ts

```diff
@@ -1,17 +1,18 @@
 // Function to normalize Arabic text
 export function normalizeArabic(text: string): string {
-	// Replace all occurrences of "أ" and "إ" with "ا"
-	text = text
-		.replace(
-			/\u0670|\u0671|\u0672|\u0673|\u0675|\u0622|\u0623|\u0625|\u0616|\u0656/g,
-			"ا",
-		)
-		.replace(/ /g, "ا")
-
-		.replace(/[^^\u0621-\u064A]/g, ""); // Keep only Arabic letters
-
-	// Remove diacritics (شكل) and hamza (همزة)
-	text = text.replace(/[ؐ-ًؕ-ٖٓ-ٟۖ-ۭ]/g, ""); // Remove diacritics
-
-	return text;
+    return text
+        // Remove diacritics (tashkeel)
+        .replace(/[\u0617-\u061A\u06d6\u06d7\u064B-\u0652]/g, "")
+        // Remove small alef (alef khanjariya)
+        .replace(/\u0670/g, "")
+        // Normalize alef variants
+        .replace(/[\u0622\u0623\u0625]/g, "\u0627")
+        // Normalize teh marbuta
+        .replace(/\u06C3/g, "\u0629")
+        // Normalize waw variants
+        .replace(/\u0624/g, "\u0648")
+        // Remove extended arabic letters
+        .replace(/[\u06D5\u06EE\u06EF]/g, "")
+        // Remove small kasra
+        .replace(/\u061A/g, "");
```

Comment:
```suggestion
return (
    text
      // Remove diacritics (tashkeel)
      .replace(/[\u0617-\u061A\u06d6\u06d7\u064B-\u0652]/g, "")
      // Remove small alef (alef khanjariya)
      .replace(/\u0670/g, "")
      // Normalize alef variants
      .replace(/[\u0622\u0623\u0625]/g, "\u0627")
      // Normalize taa marbuta
      .replace(/\u06C3/g, "\u0629")
      // Normalize waw variants
      .replace(/\u0624/g, "\u0648")
      // Remove extended arabic letters
      .replace(/[\u06D5\u06EE\u06EF]/g, "")
      // Remove small kasra
      .replace(/\u061A/g, "")
      // Remove any remaining double spaces
      .replace(/\s+/g, " ")
  );
```

Adding the last replace to remove any consequent spaces (like when there is `ۗ`).